sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterValidationInAsyncShouldBeWrapped.cs

```diff
@@ -25,19 +27,36 @@ public class InvalidCases
 
             if (value == null)
             {
-                throw new ArgumentNullException(nameof(value));
+                throw new ArgumentNullException(nameof(value)); // Secondary
             }
         }
 
         public async void OnSomeEvent(object sender, EventArgs args) // Noncompliant - it might looks weird to throw from some event method but that's valid syntax
         {
             if (sender == null)
             {
-                throw new ArgumentNullException(nameof(sender));
+                throw new ArgumentNullException(nameof(sender)); // Secondary
             }
 
             await Task.Yield();
         }
+
+        public static Task SkipLinesAsync(this TextReader reader, int linesToSkip) // Noncompliant - Not sure what should be the expected result (i.e. do local functions suffer from this behavior?)
```

Comment:
That is nice. Perhaps one more test when the exceptions ale only thrown in local functions and not in the method?